src/rp/util/hash.clj

```diff
@@ -0,0 +1,10 @@
+(ns rp.util.hash
+  (:import [java.security MessageDigest]
+           [java.math BigInteger]))
+
+(defn md5
+  [^String s]
+  (let [algorithm (MessageDigest/getInstance "MD5")
+        raw (.digest algorithm (.getBytes s))
```

Comment:
No reflection is required as-is; it would be required without the `^String` type hint.